### PR TITLE
Issues/#20 custom fonts resources

### PR DESCRIPTION
**Pull request by DevTchernov**

### CODE REVIEW

File: gradle-plugin/src/main/kotlin/dev/icerock/gradle/MultiplatformResourcesPlugin.kt [hunk 1]
```diff
@@ -110,15 +88,25 @@ class MultiplatformResourcesPlugin : Plugin<Project> {
...
+                androidPackage
+            )
+            val features = with(sourceInfo) {
+                listOf(
+                    StringsGeneratorFeature(this),
+                    PluralsGeneratorFeature(this),
+                    ImagesGeneratorFeature(this),
+                    FontsGeneratorFeature(this)
+                )
+            }
```
Comment (Alex009) [inline gradle-plugin/src/main/kotlin/dev/icerock/gradle/MultiplatformResourcesPlugin.kt]: i think
```kotlin
val features = listOf(
    StringsGeneratorFeature(sourceInfo),
    PluralsGeneratorFeature(sourceInfo),
    ImagesGeneratorFeature(sourceInfo),
    FontsGeneratorFeature(sourceInfo)
)
```
more readable in this case

File: gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/fonts/FontsGenerator.kt [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        val spec = TypeSpec.objectBuilder(familyName)
+        fontStyleFiles
+            .map { Pair(it.first, getPropertyInitializer(it.second)) }
+            .filter { it.second != null }
+            .forEach { (styleName, codeBlock) ->
+                val styleProperty = PropertySpec
+                    .builder(styleName, resourceClass)
+                    .initializer(codeBlock!!)
+                spec.addProperty(styleProperty.build())
+            }
```
Comment (Alex009) [inline gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/fonts/FontsGenerator.kt]: without naming it's difficult to understand what here doing

File: gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/fonts/IosFontsGenerator.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+) : FontsGenerator(
+    sourceSet = sourceSet,
+    inputFileTree = inputFileTree
+) {
+    override fun getClassModifiers(): Array<KModifier> = arrayOf(KModifier.ACTUAL)
+
+    override fun getPropertyModifiers(): Array<KModifier> = arrayOf(KModifier.ACTUAL)
+
+    override fun getPropertyInitializer(fontFileName: String): CodeBlock? {
+        return CodeBlock.of("FontResource(fontName = %S", fontFileName)
```
Comment (Alex009) [inline gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/fonts/IosFontsGenerator.kt]: it's generate string like `FontResource(fontName = "Roboto.ttf"`. you miss `)`

**Comment by Alex009** [comment]
@DevTchernov target branch should be `develop`

File: gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/fonts/FontsGenerator.kt [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            key.substringBefore("-")
+        }
+
+        familyGroups.forEach{ group ->
+            var stylePairs = group
+                .value
+                .map { Pair(
+                    it.substringAfter("-"),
+                    it)
+                }
```
Comment (Alex009) [inline gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/fonts/FontsGenerator.kt]: reformat file from idea.
https://www.jetbrains.com/help/idea/reformat-and-rearrange-code.html

File: gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/fonts/IosFontsGenerator.kt [hunk 5]
```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2019 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
+ */
```
Comment (Alex009) [inline gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/fonts/IosFontsGenerator.kt]: now 2020 year :)

File: gradle-plugin/src/main/kotlin/dev/icerock/gradle/MultiplatformResourcesPlugin.kt (hunk 1 above)
Comment (DevTchernov) [reply gradle-plugin/src/main/kotlin/dev/icerock/gradle/MultiplatformResourcesPlugin.kt]: Done

File: gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/fonts/IosFontsGenerator.kt (hunk 3 above)
Comment (DevTchernov) [reply gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/fonts/IosFontsGenerator.kt]: Fixed

File: gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/fonts/FontsGenerator.kt (hunk 2 above)
Comment (DevTchernov) [reply gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/fonts/FontsGenerator.kt]: Fixed

File: gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/fonts/FontsGenerator.kt (hunk 4 above)
Comment (DevTchernov) [reply gradle-plugin/src/main/kotlin/dev/icerock/gradle/generator/fonts/FontsGenerator.kt]: Done

**Pull request merged**